tests/positive/luks/creation.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package luks
+
+import (
+	"github.com/coreos/ignition/v2/tests/register"
+	"github.com/coreos/ignition/v2/tests/types"
+)
+
+func init() {
+	register.Register(register.PositiveTest, LuksWithStaticKey())
+	register.Register(register.PositiveTest, LuksWithTPM())
+
+}
+
+func LuksWithStaticKey() types.Test {
+	name := "static.filesystem.luks.encrypt"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	mntDevices := []types.MntDevice{
+		{
+			Label:        "OEM",
+			Substitution: "$DEVICE",
+		},
+	}
+	config := `{
+		"ignition": { "version": "$version" },
+		"storage": {
+		  "disks": [
+			{
+			  "device": "$DEVICE",
+			  "partitions": [
+				{
+					"label": "luks-device",
+					"sizeMiB": 0,
+					"startMiB": 0
+				}
+				],
+				"wipe_table": true
+			}
+		  ],
+		  "luks": [
+			{
+			  "device": "$DEVICE",
+			  "name": "$UUID",
+			  "keyFile": {
+				"compression": "",
+				"source": "data:,REPLACE-THIS-WITH-YOUR-KEY-MATERIAL"
+			  },
+			  "wipe_volume": true
+			}
+		  ]
+		}
```

Comment:
This test is creating a partition on `$DEVICE`, but then creating a LUKS device on the disk instead of the partition, nuking the freshly created partition table. Is that intended?

Similarly below.

Reply:
Yeah; Some how doing it this way made the tests pass. I am not sure why but I was using 'wipe_volume' which is bu. not ign. and this config got away with it. I noticed that just now and dropped the partitions and switched to the correct .ign field.